dynamic_programming/fibonacci_memoization.cpp

```diff
@@ -0,0 +1,96 @@
+/**
+ * @file
+ * @brief Implementation of [nth Fibonacci problem](https://medium.com/@andrewsouthard1/dynamic-programming-recursion-memoization-and-bottom-up-algorithms-61c882d1c7e) using memoization
+ * @details
+ * nth Fibonacci number using memoization is used as a dynamic programming concept that
+ * works in \f$O(n)\f$ time and works in \f$O(n)\f$ space.
+ * Other solution is bottom up approach which uses \f$O(1)\f$ space. 
+ * @author [Pradyumna Yadav](https://github.com/pradyyadav)
+ */
+
+#include <cassert>            /// for assert
+#include <iostream>          /// for io operations
+#include <cstring>           /// for memset
+
+/**
+ * @namespace dynamic_programming
+ * @brief Dynamic Programming algorithms
+ */
+namespace dynamic_programming {
+/**
+ * @namespace fibonacci_memoization
+ * @brief Functions for the [nth Fibonacci](https://medium.com/@andrewsouthard1/dynamic-programming-recursion-memoization-and-bottom-up-algorithms-61c882d1c7e) using memoization implementation
+ */
+namespace fibonacci_memoization {
+/**
+ * @brief The function that implements the nth Fibonacci number problem using memoization.
+ * @param n The index of the number in a Fibonacci series.
+ * @param term
+ * @returns Element at the nth position in the Fibonacci series.
+ */
+long long int nthFibonacci(int n, int term[])
```

Comment:
Consider using `uint64_t` for non-negative values (or their appropriate size: `uint32_t`, `uint16_t`, `uint8_t`) or `int64_t` for negative values. Check other parts of the code ([reference](https://en.cppreference.com/w/cpp/types/integer)). **Let us know if you need any help**. 🙂

Reply:
Thank you, actually when I am using command: ```clang-tidy --fix --quiet -p build dynamic_programming/fibonacci_memoization.cpp --``` inside my local repo folder, it is throwing this error: ```clang-tidy: The term 'clang-tidy' is not recognized as a name of a cmdlet, function, script file, or executable program.
Check the spelling of the name, or if a path was included, verify that the path is correct and try again.``` I know it is due to some path error. Can you please tell me what's wrong here.